Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx

```diff
@@ -100,33 +100,30 @@ const partitionRegistrations = (registrations) => {
 
 export default function RegistrationAdministrationList() {
   const { competition_id } = useParams()
-  const [registrations, setRegistrations] = useState([])
-  const [isLoading, setIsLoading] = useState(true)
+  const {
+    isLoading,
+    data: registrations,
+    refetch,
+  } = useQuery({
+    queryKey: ['registrations-admin', competition_id],
+    queryFn: () => getAllRegistrations(competition_id),
+  })
   const [selected, dispatch] = useReducer(reducer, {
     waiting: [],
     accepted: [],
     deleted: [],
   })
-  const fetchData = async (competition_id) => {
-    const registrations = await getAllRegistrations(competition_id)
-    const regList = []
-    for (const registration of registrations) {
-      registration.user = (await getCompetitorInfo(registration.user_id)).user
-      regList.push(registration)
-    }
-    return regList
-  }
-  useEffect(() => {
-    fetchData(competition_id).then((registrations) => {
-      setRegistrations(registrations)
-      setIsLoading(false)
-    })
-  }, [competition_id])
 
-  const { waiting, accepted, deleted } = useMemo(
-    () => partitionRegistrations(registrations),
-    [registrations]
-  )
+  const { waiting, accepted, deleted } = useMemo(() => {
+    if (registrations) {
+      return partitionRegistrations(registrations)
+    }
+    return {
+      waiting: [],
+      accepted: [],
+      deleted: [],
+    }
```

Comment:
I think `() => partitionRegistrations(registrations ?? [])` would work, though maybe having the still-loading case explicit is nice.